AIDA FAQ (#6711)
harness/developer-hub: kb/continuous-delivery/continuous-delivery-faqs.md

For more information, go to [Looping Strategies](https://developer.harness.io/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism/#repeat-on-multiple-target-hosts)

#### Can a user create/update a confluence page from a Harness step?
No, currently there is no native step to create or update a confluence page.
#### How do I run a stand-alone deployment?
To deploy a single instance of an application or software component independently in Harness, you can use the Basic deployment strategy.

> #### How to install PowerShell on harness delegate?

Reviewer: ```suggestion
#### How to install PowerShell on Harness Delegate?
```

> You can run PowerShell scripts on a Harness Delegate, even though the Delegate must be run on Linux. Linux supports PowerShell using PowerShell core. https://developer.

Reviewer: ```suggestion
You can run PowerShell scripts on a Harness Delegate, even though the delegate must be run on Linux. Linux supports PowerShell using PowerShell core. For more details, go to [Common delegate profile scripts](https://developer.harness.io/docs/first-gen/firstgen-platform/techref-category/account-ref/delegate-ref/common-delegate-profile-scripts/#powershell).
```

> 
> harness.io/docs/first-gen/firstgen-platform/techref-category/account-ref/delegate-ref/common-delegate-profile-scripts/#powershell 

Reviewer: ```suggestion
```

```
# Download the Microsoft repository GPG keys  
wget -q https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb  
  
# Register the Microsoft repository GPG keys  
sudo dpkg -i packages-microsoft-prod.deb  
  
# Update the list of products  
sudo apt-get update  
  
# Install PowerShell  
sudo apt-get install -y powershell  
  
# Start PowerShell  
pwsh

```

#### Could you please provide the documentation or guidelines for establishing audit logs to monitor technology/device activity and file integrity monitoring?

> https://developer.harness.io/docs/platform/governance/audit-trail/audit-streaming/ and https://developer.harness.io/docs/platform/governance/audit-trail/

Reviewer: ```suggestion
* [Audit streaming](https://developer.harness.io/docs/platform/governance/audit-trail/audit-streaming/)
* [Audit trail](https://developer.harness.io/docs/platform/governance/audit-trail/)
```

Some potential reasons for this issue could include:

1. Incorrect repository URL or credentials: Double-check that the repository URL and credentials provided in the Harness configuration are accurate and have the necessary permissions to access the repository.

2. Incorrect file path or name: Ensure that the file path specified in the Harness configuration matches the actual location and name of the manifest file in the Bitbucket repository.

3. Incorrect branch or tag specified: Verify that the branch or tag specified in the Harness configuration corresponds to the correct version of the manifest file in the Bitbucket repository.

4. Issues with the Bitbucket API: If there are any issues or downtime with the Bitbucket API, it could prevent Harness from fetching the manifest file. Check the status of the Bitbucket API and any error messages in the Harness logs for more information.

It's recommended to review the Harness logs and Bitbucket API logs for any error messages or additional details that could help pinpoint the root cause of the issue.

#### What is the expression for the branch name?

> In the Trigger scenario, you can utilize ```<+trigger.targetBranch>``` if Git Experience is enabled for your pipeline. This field represents the PR's target branch or the branch specified for a branch build. Additionally, in the Pipeline Reference Branch field, you can use ```<+trigger.branch>``` to refer to the same target branch.

Reviewer: ```suggestion
In the trigger scenario, you can utilize ```<+trigger.targetBranch>``` if GitEx is enabled for your pipeline. This field represents the PR's target branch or the branch specified for a branch build. Additionally, in the Pipeline Reference Branch field, you can use ```<+trigger.branch>``` to refer to the same target branch.
```

> #### Is it supported to use environment-level ```<+input>``` fields when running a pipeline?

Reviewer: ```suggestion
#### Are environment-level ```<+input>``` fields supported when running a pipeline?
```

> Yes, it is supported to use environment-level ```<+input>``` fields when running a pipeline. These runtime inputs can be defined in the environment and then selected at runtime when running the pipeline.

Reviewer: ```suggestion
Yes, Harness supports using environment-level ```<+input>``` fields when running a pipeline. These runtime inputs can be defined in the environment and then selected at runtime when running the pipeline.
```

#### How to Install the AAD Pod Identity components on your AKS cluster using the Azure CLI or Helm chart?
To install the Azure Active Directory (AAD) Pod Identity components on your Azure Kubernetes Service (AKS) cluster, you have two options: using the Azure CLI or Helm chart.

Using Azure CLI:

> * Install AAD Pod Identity:

Reviewer: ```suggestion
1. Install AAD Pod Identity:
```

```
az aks pod-identity add --resource-group <resource-group-name> --cluster-name <cluster-name>
```

> * Verify the installation:

Reviewer: ```suggestion
2. Verify the installation:
```

> * Install AAD Pod Identity:
> ```
> az aks pod-identity add --resource-group <resource-group-name> --cluster-name <cluster-name>
> ```
> 
> * Verify the installation:
> ```kubectl get pods -n kube-system
> ```

Reviewer: ```suggestion
1. Install AAD Pod Identity:
    ```
    az aks pod-identity add --resource-group <resource-group-name> --cluster-name <cluster-name>
   ```

2. Verify the installation:
    ```
    kubectl get pods -n kube-system
   ```
```

Using Helm Chart:

> * Add the Helm repository:

Reviewer: ```suggestion
1. Add the Helm repository:
```

```
helm repo add aad-pod-identity https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts
```

> * Install AAD Pod Identity:

Reviewer: ```suggestion
2. Install AAD Pod Identity:
```

```
helm install aad-pod-identity aad-pod-identity/aad-pod-identity
```

> * Verify the installation:

Reviewer: ```suggestion
3. Verify the installation:
```

```
kubectl get pods -n kube-system
```

These are the steps to install AAD Pod Identity components on your AKS cluster using either the Azure CLI or Helm chart. Replace ```<resource-group-name> ``` and ``` <cluster-name> ``` with your actual resource group and cluster name, respectively.

> #### Write an OPA policy to check if the production stage conditional execution is set to always in Harness

Reviewer: ```suggestion
#### Write an OPA policy to check if the production stage conditional execution is set to always in Harness.
```

> Below is an example of an OPA policy to check if the conditional execution for the production stage is set to "always" in Harness:

Reviewer: ```suggestion
Below is an example OPA policy to check if the conditional execution for the production stage is set to "always" in Harness:
```

```
package harness_production_stage_check

default allow = false

production_stage_condition {
    stage := input.request.body.stage
    stage_name := stage.name
    stage_type := stage.type
    stage_details := stage.details
    stage_type == "PRODUCTION"   // Check if the stage is a production stage
    stage_details.conditional_execution == "always"  // Check if the conditional execution is set to "always"
}

allow {
    production_stage_condition
}
```
This policy checks if the stage is a production stage and if its conditional execution is set to "always". If both conditions are met, it allows the request. Otherwise, it denies the request. 

> #### Could you please clarify What is the message of this pop-up "input set does not exist in the selected branch"?

Reviewer: ```suggestion
#### Could you please clarify what is the message of this pop-up "input set does not exist in the selected branch"?
```

If the input set you're trying to use isn't found in the branch where the pipeline is located, Harness will display this message.

#### Why It doesn't permit us to deploy a WinRM type service because we cannot configure the artifact to be installed on the destination server.
Currently, you cannot add artifacts to your Harness WinRM service. This is because the WinRM deployment type does not permit configuring the artifact to be installed on the destination server. 

Instead, you can use the Command step to run commands on the target hosts to download or copy the artifact to the destination server.

#### How does the rollback process work in SSH deployments?
Harness restores the state of deployment to the pipeline's previous successful stage execution based on serivce, enviroment and infrastucture details. Harness records the artifact version that was successfully deployed during previous successful executions. When using the Rollback step's Copy Artifact command unit, Harness copies the last successful version of the artifact deployed via Harness to the remote host.

> https://developer.harness.io/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng/#rollback

Reviewer: ```suggestion
For more details, go to [SSH deployments](https://developer.harness.io/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng/#rollback).
```

#### How to avoid Harness pulling old Helm dependencies that are not listed in the Chart.yaml file?
To avoid this issue, ensure that you are using a delegate version above 818xx.

> #### Is it expected behaviour for the environment dropdown to display all environments when there's no mapping associated with the environment group?

Reviewer: ```suggestion
#### Is it expected behaviour for the environment drop-down to display all environments when there's no mapping associated with the environment group?
```

> Yes, that is the expected behaviour. When there's no mapping associated with the environment group, all environments will be displayed in the dropdown menu.

Reviewer: ```suggestion
Yes, that is the expected behaviour. When there's no mapping associated with the environment group, all environments will be displayed in the drop-down menu.
```

#### What does redeploy do?

> The "redeploy" action in Harness refers to the process of restarting or relaunching an application or service that is already deployed. This action ensures that any changes made to the application's configuration or underlying infrastructure are applied and that the application is running with the most up-to-date settings.

Reviewer: ```suggestion
The "redeploy" action in Harness refers to the process of restarting or relaunching an application or service that is already deployed. This action ensures that any changes made to the application's configuration or underlying infrastructure are applied and the application is running with the most up to date settings.
```
